packages/rule-tester/src/RuleTester.ts

```diff
@@ -1030,6 +1050,73 @@ function assertASTDidntChange(beforeAST: unknown, afterAST: unknown): void {
   assert.deepStrictEqual(beforeAST, afterAST, 'Rule should not modify AST.');
 }
 
+/**
+ * Check if a value is a primitive or plain object created by the Object constructor.
+ */
+function isSerializablePrimitiveOrPlainObject(val: unknown): boolean {
+  return (
+    // eslint-disable-next-line eqeqeq
+    val === null ||
+    typeof val === 'string' ||
+    typeof val === 'boolean' ||
+    typeof val === 'number' ||
+    (typeof val === 'object' && val.constructor === Object) ||
+    Array.isArray(val)
+  );
+}
+
+/**
+ * Check if a value is serializable.
+ * Functions or objects like RegExp cannot be serialized by JSON.stringify().
+ * Inspired by: https://stackoverflow.com/questions/30579940/reliable-way-to-check-if-objects-is-serializable-in-javascript
+ */
+function isSerializable(val: unknown): boolean {
+  if (!isSerializablePrimitiveOrPlainObject(val)) {
+    return false;
+  }
+  if (typeof val === 'object') {
+    const valAsObj = val as Record<string, unknown>;
+    for (const property in valAsObj) {
+      // TODO: use `Object.hasOwn` (used in eslint@9) once we upgrade to eslint@9
```

Comment:
[Non-Actionable] Oh! I bet we can do this actually. If ESLint is using a new JS built-in then that's a sign that their language support range must include it.

Filed #9028.

```suggestion
      // TODO(#9028): use `Object.hasOwn` (used in eslint@9) once we upgrade to eslint@9
```